proposals/NNNN-unlock-existentials-with-fixed-associated-types.md

```diff
@@ -0,0 +1,236 @@
+# Unlock Existentials with Fixed Associated Types
+
+* Proposal: [SE-NNNN](NNNN-unlock-existentials-with-fixed-associated-types.md)
+* Authors: [Filip Sakel](https://github.com/filip-sakel), [Anthony Latsis](https://github.com/AnthonyLatsis)
+* Review Manager: TBD
+* Status: **Awaiting implementation**
+
+## Introduction
+
+Today, Swift protocols are divided into two categories: those that _can_ and those that _can't_ be used as Types. The latter can only be used as generic constraints and _not_ as Types - because they have 'Self' or associated type requirements. However, in some cases where a protocol inherits and fixes the associated types of another protocol - that doesn't have `Self` requirements, this constraint seems rather unintuitive. This proposal aims to relax this needless contraint allowing such protocols to be used as Types.
```

Comment:
I agree that the Introduction section should be short and that the motivational argument best belongs elsewhere. Nevertheless, mentioning the general solution whereof this proposal is a part, seems to belong to a different section such as a Background section rather than the Introduction one. I think that with such a complex topic including such a= section would be useful in informing the reader how this proposal fits in with the more general goal of Generalized Existentials. Do you think that a Background section would be a good addition?

Reply:
> I agree that the Introduction section should be short and that the motivational argument best belongs elsewhere. Nevertheless, mentioning the general solution whereof this proposal is a part, seems to belong to a different section such as a Background section rather than the Introduction one.

I was mentioning it as a means of introducing the problem, because the proposal is likely to become one in a series of incremental efforts towards the collective concept of generalized existentials. This means that generalized existentials is a long-term goal for this proposal, rather than an *orthogonal* future direction. I will not mind if you still choose to omit it here, though.

> Do you think that a Background section would be a good addition?

Is this not the responsibility of the future directions section?

Reply:
**Is this not the responsibility of the future directions section?**

I Agree.

**...is likely to become one in a series of incremental efforts towards the collective concept of generalized existentials.**

That's the goal! And since it's a goal that entails future additions, I think that the Future Directions section would best capture the overall direction the Language is heading towards while also laying out some steps we deem noteworthy (what's discussed in the section).

Reply:
> That's the goal! And since it's a goal that entails future additions, I think that the Future Directions section would best capture the overall direction the Language is heading towards while also laying out some steps we deem noteworthy (what's discussed in the section).

On the other hand, it also seems like an omission to only start mentioning its direct relation to generalized existentials at the very end.

Reply:
> Nevertheless, mentioning the general solution whereof this proposal is a part, seems to belong to a different section such as a Background section rather than the Introduction one

FYI, this is exactly what the [Opaque Result Types](https://github.com/apple/swift-evolution/blob/master/proposals/0244-opaque-result-types.md) proposal does. Mentioning (not explaining) the superior topic is a way to provide context early on to those who already are in the subject and let others familiarize themselves before reading on to grasp the specific problem the proposal addresses.